blog/2025-05-07-nanite-at-home/index.md

```diff
@@ -0,0 +1,172 @@
+
+I don't think that implementing Nanite is worth it from a pure performance standpoint, except maybe for AAA studios who have the resources to build it and get it to a good enough quality. But if you want pure performance, you should first focus on getting all the basics out of the way. Like I haven't implemented any form of frustum culling, no two-pass occlusion culling, still use a G-Buffer instead of a [Visibility buffer](http://filmicworlds.com/blog/visibility-buffer-rendering-with-material-graphs/) cause it was faster to implement, could likely benefit from batching my draws and should compress my vertex attributes better. All of these will likely have better performance yields for the time invested than implementing Nanite.
+
+I could see Nanite as a tool for artists though, removing their need to manually create LODs and save money in production. However, I can't reasonably give you any recommendations, I'm not an artist myself nor do I know how much time is spent on creating LODs. But there seems to be an active need in that area, since there's multiple companies selling software just for mesh preprocessing and simplification.
+
+Closing it out, if you're curious and would like more detail on how nanite works, feel free to download my master thesis and have a read! And yes, that means I'm a recent graduate (with some work experience as well) looking for a Job, so feel free to contact me on [my github](https://github.com/Firestar99). I also want to give a huge shoutout to [jglrxavpok](https://jglrxavpok.github.io/) and [jms55](https://jms55.github.io/posts/) who have built their own Nanite engines and documented some of their work, which has helped me a lot! And [zeux](https://github.com/zeux) for the amazing [meshoptimizer](https://github.com/zeux/meshoptimizer/) library and their own [Nanite experiments](https://github.com/zeux/meshoptimizer/blob/master/demo/nanite.cpp) to reference!
+
+Maybe I'll even make another article soon, introducing my bindless system for rust-gpu, designed similarly to [Daxa](https://github.com/Ipotrick/Daxa), and how to make a gpu-driven renderer with it. But first, I'd like to see [cargo gpu](https://github.com/Rust-GPU/cargo-gpu/) getting production ready, to give a small teaser into upcoming rust-gpu tooling.
```

Comment:
I think these should be alluded to in the intro and these are the real novel bits.